notifyme-sdk-backend/notifyme-sdk-backend-ws/src/main/java/ch/ubique/notifyme/sdk/backend/ws/insertmanager/insertfilters/OverlappingIntervalsFilter.java

```diff
@@ -0,0 +1,48 @@
+package ch.ubique.notifyme.sdk.backend.ws.insertmanager.insertfilters;
+
+import ch.ubique.notifyme.sdk.backend.model.UserUploadPayloadOuterClass;
+import ch.ubique.notifyme.sdk.backend.ws.insertmanager.InsertException;
+import ch.ubique.notifyme.sdk.backend.ws.insertmanager.OSType;
+import ch.ubique.notifyme.sdk.backend.ws.semver.Version;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+/**
+ * A person can't be in two places at the same time. UploadVenueInfo objects whose time windows
+ * overlap are both removed.
+ */
+public class OverlappingIntervalsFilter implements UploadInsertionFilter {
```

Comment:
@wouterl is the idea to filter out overlapping events or reject the full request as soon as we have one overlap in there? I would say the second one, because the app should make sure it never sends overlaps, so a request will be malicious if there is an overlap in there and we should reject all of that request.